services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/RestPreconditions.java

```diff
@@ -0,0 +1,118 @@
+package com.futurewei.alcor.dataplane.utils;
+
+import com.futurewei.alcor.common.entity.CustomerResource;
+import com.futurewei.alcor.common.exception.ParameterNullOrEmptyException;
+import com.futurewei.alcor.common.exception.ParameterUnexpectedValueException;
+import com.futurewei.alcor.common.exception.ResourceNotFoundException;
+import com.futurewei.alcor.common.exception.ResourceNullException;
+import com.futurewei.alcor.common.logging.Logger;
+import com.futurewei.alcor.common.logging.LoggerFactory;
+import com.futurewei.alcor.dataplane.config.Config;
+import org.thymeleaf.util.StringUtils;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.logging.Level;
+
+import static com.futurewei.alcor.schema.Subnet.SubnetState;
+import static com.futurewei.alcor.schema.Vpc.VpcState;
+//TODO leave it for now
+public class RestPreconditions {
```

Comment:
Most of methods in RestPreconditions have been moved to AlcorCommon. We can reuse the methods there except that DPM has its specified check.

Reply:
would remove it